packages/uiweb/src/lib/components/chat/MessageBubble/MessageBubble.tsx

```diff
@@ -73,101 +137,273 @@ const FileCard = ({
     const size = fileContent.size;
 
     return (
-        <Section
-            alignSelf={position ? 'end' : 'start'}
-            maxWidth="80%"
-            margin="5px 0"
-            background="#343536"
-            borderRadius="8px"
-            justifyContent="space-around"
-            padding="10px 13px"
-            gap="15px"
-        >
-            <Image
-                src={FILE_ICON(name.split('.').slice(-1)[0])}
-                alt="extension icon"
-                width="20px"
-                height="20px"
-            />
-            <Section flexDirection="column" gap="5px">
-                <Span color="#fff" fontSize="15px">
-                    {shortenText(name, 11)}
-                </Span>
-                <Span color="#fff" fontSize="12px">
-                    {formatFileSize(size)}
-                </Span>
-            </Section>
-            <FileDownloadIconAnchor
-                href={content}
-                target="_blank"
-                rel="noopener noreferrer"
-                download
-            >
-                <FileDownloadIcon className="fa fa-download" aria-hidden="true" />
-            </FileDownloadIconAnchor>
+        <Section justifyContent="start">
+            {isGroup ? (
+                <Section flexDirection="column">
+                    {chat.fromCAIP10.split(":")[1] !== account && (
+                        <Span alignSelf="start"
+                            textAlign="left">{chat.fromDID.split(":")[1].slice(0, 6)}...
+                            {chat.fromDID.split(":")[1].slice(-6)}</Span>
+                    )}
```

Comment:
same applies here. The  code for message for both group and normal chat can be reused. it can also be done with ImageCard and all other cards